cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java

```diff
@@ -611,7 +608,19 @@ public void gitPullOrClone(Path target, String gitRepoUrl) {
         askToContinue(message);
       } else {
         pc.errorHandling(ProcessErrorHandling.WARNING);
-        result = pc.addArg("pull").run(false);
+        if (isOnline()) {
+          result = pc.addArg("fetch").addArg("origin").addArg("master").run(false);
+          if (!result.isSuccessful()) {
+            warning("Git failed to fetch from origin master");
+          }
+        }
+        if (force) {
+          result = pc.addArg("reset").addArg("--hard").addArg("HEAD~").run(false);
+          if (!result.isSuccessful()) {
+            warning("Git failed to reset to head");
```

Comment:
same here. See #625 where `target` is added to the end-user message to be more helpful.

Reply:
aha, and wenn the internet connection is interrupted shouldn't we log some warning at least?

Reply:
Not sure if we can detect that properly without using capture.

Reply:
I've added the target path now.